diff --git a/index.d.ts b/index.d.ts
index 0dd7c4f97a0cf4779fadae69a14555293351c9ec..d3520360143e7f207c8f5f326a880d8345ddd738 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -5,13 +5,13 @@ declare module 'react-helmet-async' {
     [key: string]: string | number | boolean | null | undefined;
   }
 
-  type HtmlProps = JSX.IntrinsicElements['html'] & OtherElementAttributes;
+  type HtmlProps = React.JSX.IntrinsicElements['html'] & OtherElementAttributes;
 
-  type BodyProps = JSX.IntrinsicElements['body'] & OtherElementAttributes;
+  type BodyProps = React.JSX.IntrinsicElements['body'] & OtherElementAttributes;
 
-  type LinkProps = JSX.IntrinsicElements['link'];
+  type LinkProps = React.JSX.IntrinsicElements['link'];
 
-  type MetaProps = JSX.IntrinsicElements['meta'];
+  type MetaProps = React.JSX.IntrinsicElements['meta'];
 
   export interface HelmetTags {
     baseTag: Array<any>;
